File: Backend/src/main/java/de/ali/shitpostbot/Image/model/Image.java
package de.ali.shitpostbot.Image.model;

import de.ali.shitpostbot.User.model.User;
import de.ali.shitpostbot.shared.model.BaseEntity;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import java.io.Serializable;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Image extends BaseEntity implements Serializable {
    private String url;

    @ManyToOne
    @JoinColumn(nullable = false)
    @EqualsAndHashCode.Exclude
    private User poster;
}

File: Frontend/src/shared/image/model/Image.ts
import {BaseEntity} from '../../models/BaseEntity';
import {User} from '../../user/model/User';

export interface Image extends BaseEntity {
  url: string;
  poster: User;
}

/*export class Image extends BaseEntity {
  public url: string;
  public poster: User;
  /*public set url(url: string) {
    this._url = url;
  }
  public get url(): string {
    return this._url;
  }
  public set poster(poster: User) {
    this._poster = poster;
  }
  public get poster(): User {
    return this._poster;
  }
}*/
